### PR TITLE
Pre-populate ui control with user data from AWS server group being cl…

### PR DESCRIPTION
**Pull request by duftler**
…oned.

### CODE REVIEW

File: app/scripts/modules/amazon/serverGroup/configure/serverGroupCommandBuilder.service.js [hunk 1]
```diff
@@ -244,6 +244,9 @@ module.exports = angular.module('spinnaker.aws.serverGroupCommandBuilder.service
             instanceMonitoring: serverGroup.launchConfig.instanceMonitoring.enabled,
             ebsOptimized: serverGroup.launchConfig.ebsOptimized,
           });
+          if (!!serverGroup.launchConfig.userData) {
```
Comment (duftler) [inline app/scripts/modules/amazon/serverGroup/configure/serverGroupCommandBuilder.service.js]: The linter complains. @anotherchrisberry what's the right check to use here?

**Comment by duftler** [comment]
Ok, fixed. PTAL.

**Comment by anotherchrisberry** [comment]
@duftler What's the relationship between this change and https://github.com/spinnaker/clouddriver/pull/372 ? 

Seems like the Clouddriver change would supersede the need for this change.

**Comment by anotherchrisberry** [comment]
I ask because this turns out to be a breaking change at Netflix, and I need to figure out if we should roll it back or provide a feature flag to disable it.

**Comment by duftler** [comment]
The goal with this PR and the other one is to treat user data the same way we treat the other attributes; pre-populate the ui with the ancestor value, and if the attribute is completely omitted from the request (the request to clouddriver) use the ancestor value.

**Pull request merged**